Analysis/Tasks/PWGHF/HFJpsiToEECandidateSelector.cxx

```diff
@@ -0,0 +1,235 @@
+      return false; //cut on daughter pT
+    }
+    if (TMath::Abs(trackElectron.dcaPrim0()) > cuts[pTBin][1] || TMath::Abs(trackPositron.dcaPrim0()) > cuts[pTBin][1]) {
+      return false; //cut on daughter dca - need to add secondary vertex constraint here
+    }
+    if (TMath::Abs(trackElectron.dcaPrim1()) > cuts[pTBin][2] || TMath::Abs(trackPositron.dcaPrim1()) > cuts[pTBin][2]) {
+      return false; //cut on daughter dca - need to add secondary vertex constraint here
+    }
+
+    return true;
+  }
+
+  /// Check if track is ok for TPC PID
+  /// \param track is the track
+  /// \note function to be expanded
+  /// \return true if track is ok for TPC PID
+  template <typename T>
+  bool validTPCPID(const T& track)
+  {
+    if (TMath::Abs(track.pt()) < d_pidTPCMinpT || TMath::Abs(track.pt()) >= d_pidTPCMaxpT) {
+      return false;
+    }
+    //if (track.TPCNClsFindable() < d_TPCNClsFindablePIDCut) return false;
+    return true;
+  }
+
+  /// Check if track is compatible with given TPC Nsigma cut for a given flavour hypothesis
+  /// \param track is the track
+  /// \param nPDG is the flavour hypothesis PDG number
+  /// \param nSigmaCut is the nsigma threshold to test against
+  /// \note nPDG=11 electron
+  /// \return true if track satisfies TPC PID hypothesis for given Nsigma cut
+  template <typename T>
+  bool selectionPIDTPC(const T& track, int nSigmaCut)
+  {
+    double nSigma = 1.0; //arbitarily large value
+    nPDG = TMath::Abs(11);
+    //if (nPDG == 11) {
+    //  nSigma = track.tpcNSigmaEl();
+    //} else {
+    //  return false;
+    //}
+    return nSigma < nSigmaCut;
+  }
+
+  /// PID selection on daughter track
+  /// \param track is the daughter track
+  /// \param nPDG is the PDG code of the flavour hypothesis
+  /// \note nPDG=11 electron
+  /// \return 1 if successful PID match, 0 if successful PID rejection, -1 if no PID info
+  template <typename T>
+  int selectionPID(const T& track, int nPDG)
+  {
+    int statusTPC = -1;
+    // int statusTOF = -1;
+
+    if (validTPCPID(track)) {
+      if (!selectionPIDTPC(track, d_nSigmaTPC)) {
+
+        statusTPC = 0; //rejected by PID
+      } else {
+        statusTPC = 1; //positive PID
+      }
+    } else {
+      statusTPC = -1; //no PID info
+    }
+
+    if (statusTPC == 1) {
+      return 1; //what if we have 2 && 0 ?
+    } else if (statusTPC == 0) {
+      return 0;
+    } else {
+      return -1;
+    }
+  }
+
+  void process(aod::HfCandProng2 const& hfCandProng2s, aod::BigTracksPID const& tracks)
+  {
+    int statusJpsi; // final selection flag : 0-rejected  1-accepted
+    int pidJpsi, electronPlus, electronMinus;
+
+    for (auto& hfCandProng2 : hfCandProng2s) { //looping over 2 prong candidates
+
+      auto trackPos = hfCandProng2.index0_as<aod::BigTracksPID>(); //positive daughter
+      auto trackNeg = hfCandProng2.index1_as<aod::BigTracksPID>(); //negative daughter
+
+      statusJpsi = true;
```

Comment:
Why did this change from `0` to `true` now?

Reply:
Sorry, Just a mistake here. It should be 0. I have changed back it.